python/utils/LinkedLibraryHolder.cpp

```diff
@@ -30,6 +30,30 @@ constexpr static const char NVQIR_SIMULATION_BACKEND[] =
     "NVQIR_SIMULATION_BACKEND=";
 constexpr static const char TARGET_DESCRIPTION[] = "TARGET_DESCRIPTION=";
 
+int countGPUs() {
+  char buffer[1024];
+  std::string output;
+  FILE *fp1, *fp2;
+
+  fp1 = popen("nvidia-smi", "r");
+  if (NULL == fp1) {
+    cudaq::info("nvidia-smi: command not found");
+    return -1;
+  }
+  pclose(fp1);
+
+  fp2 = popen("nvidia-smi -L | wc -l", "r");
+  if (NULL == fp2) {
```

Comment:
I would prefer you remove the use of `NULL` throughout. That's a holdover from C and considered "bad style". Thanks!

Reply:
Done.